### ISSUE TITLE
Cannot wildcard match directories / not working

### ISSUE THREAD

**Issue by Rubber-Duckie** [safing/portmaster#1552]
**What happened**:

Various applications can spawn executables that need internet access. These executables can be created in sub directories which are generated with a random name. 

Take for example visual studio, rather notorious for this, where it uses the temporary directory in which it will create subdirectories with random names, a new name every time it runs.

 i.e. 
C:\Users\User1\AppData\Local\Temp\\`uogzuxa0.jm3`\resources\app\ServiceHub\Services\Microsoft.VisualStudio.Setup.Service

This happens every few hours thus the Portmaster creates a new profile for each unique path it sees, forever. This effectively destroys the ability for the application to be of any use.

![image](https://github.com/safing/portmaster/assets/81572091/96f5d3e7-0dbc-4c3b-9743-c45c8cdf1652)

Despite having edited the profile for the executable, and having inserted a wildcard into the path within the edit profile option, it does not function as expected. 

see the screen shot below. 
![image](https://github.com/safing/portmaster/assets/81572091/32a01ba9-e366-41fd-81df-cff18fe5213e)

**What did you expect to happen?**:

Edit the profile, insert a asterisk for 'match any string' within the file path. From then on, the file path matching logic will use that single profile for the matching instances detected.

i.e. 
C:\Users\User1\AppData\Local\Temp\\`*`\resources\app\ServiceHub\Services\Microsoft.VisualStudio.Setup.Service

**Comment by Rubber-Duckie** [comment]
I notice there is a regular expression capability. 
I have merged all the duplicates into a single master. 
I have created a regex to attempt to undertake the same match. 
Will update here is this works for reference for other people with similar expectations. 
I am anticipating this will be marked expected behavour since you have NOT implemented wildcard matching strings in the fingerprint feature, and regex is the only way...
